src/share/classes/com/alibaba/rcm/internal/AbstractResourceContainer.java

```diff
@@ -181,8 +182,25 @@ public Long getResourceLimitReachedCount(ResourceType resourceType) {
 
         @Override
         public List<Long> getActiveContainerThreadIds() {
-            // root resource container is not monitored
-            return Collections.emptyList();
+            ThreadGroup group = Thread.currentThread().getThreadGroup();
+            while (group.getParent() != null) {
+                group = group.getParent();
+            }
+            int count = group.activeCount();
+            Thread[] threads;
+            do {
+                threads = new Thread[count + (count / 2) + 1]; //slightly grow the array size
+                count = group.enumerate(threads, true);
+                //return value of enumerate() must be strictly less than the array size according to javadoc
+            } while (count >= threads.length);
```

Comment:
group.activeCount() ->
Returns an **estimate** of the number of active groups in this thread group and its subgroups.
group.enumerate() -> 
Copies into the specified array references to every active subgroup in this thread group and its subgroups.

since it is a static number from that timestamp, in rare cases this could be true

Reply:
![image](https://user-images.githubusercontent.com/30545334/124588503-733b0400-de8b-11eb-85b3-eaee258c0d3a.png)
Also, it is safe from array index out of bounds